apps/ideas/views.py

```diff
@@ -57,20 +57,19 @@ class Meta:
         fields = ['category']
 
 
-class IdeaListView(mixins.ProjectMixin, FilteredListView):
-    model = idea_models.Idea
-    paginate_by = 15
-    filter_set = IdeaFilterSet
+class RequestMixin(object):
 
     def dispatch(self, *args, **kwargs):
-        # TODO: Refactor duplicate to ProjectMixin.dispatch()
-        self.project = kwargs['project']
-        self.phase = self.project.active_phase or self.project.past_phases[0]
-        self.module = self.phase.module if self.phase else None
         self.request.module = self.module
         return super().dispatch(*args, **kwargs)
 
 
+class IdeaListView(mixins.ProjectMixin, RequestMixin, FilteredListView):
```

Comment:
I do not like that you have to know the exact order of these three superclasses. Can we maybe combine `ProjectMixin` and `RequestMixin` into one?